ocs_ci/resiliency/resiliency_workload_config.py

```diff
@@ -0,0 +1,355 @@
+            dict: Configuration summary
+        """
+        return {
+            "workloads": self.get_workloads(),
+            "run_workload": self.should_run_workload(),
+            "enable_verification": self.should_run_verification(),
+            "scaling_enabled": self.is_scaling_enabled(),
+            "vdbench_threads": self.get_vdbench_threads(),
+            "vdbench_elapsed": self.get_vdbench_elapsed(),
+            "background_operations_enabled": self.get_background_operations_config().get(
+                "enabled", False
+            ),
+        }
+
+    def validate_config(self) -> bool:
+        """
+        Validate the current configuration.
+
+        Returns:
+            bool: True if configuration is valid
+
+        Raises:
+            ValueError: If configuration is invalid or missing
+        """
+        try:
+            # Check if resiliency_config exists - must be provided
+            resiliency_config = self.config.ENV_DATA.get("resiliency_config", {})
+            if not resiliency_config:
+                raise ValueError(
+                    "Missing required 'resiliency_config' section in configuration. "
+                    "Resiliency tests require explicit configuration due to their resource-intensive nature. "
+                    "Please provide configuration via --ocsci-conf parameter. "
+                    "Example: run-ci --ocsci-conf conf/ocsci/resiliency_tests_config.yaml "
+                    "See conf/ocsci/resiliency_tests_config.yaml for configuration examples."
+                )
+
+            # Validate workload names if any are specified
+            workloads = self.get_workloads()
+            if workloads:
+                # Define valid workload types
+                valid_workloads = {
+                    self.VDBENCH,
+                    self.CNV_WORKLOAD,
+                    self.FIO,
+                    self.RGW_WORKLOAD,
+                }
+
+                # Check for invalid workload names (catches typos)
+                invalid_workloads = set(workloads) - valid_workloads
+                if invalid_workloads:
+                    raise ValueError(
+                        f"Invalid workload type(s): {', '.join(invalid_workloads)}. "
+                        f"Valid workload types are: {', '.join(sorted(valid_workloads))}"
+                    )
+            else:
+                # Empty workload list is valid for failure-injection-only scenarios
+                log.info(
+                    "No workloads configured. Running in failure-injection-only mode. "
+                    "No workload will be executed during the test."
+                )
+
+            # Validate VDBENCH config if enabled
+            if "VDBENCH" in workloads:
+                vdbench_config = self.get_vdbench_config()
+                if not vdbench_config:
+                    log.warning("VDBENCH enabled but no vdbench_config found")
+                    return True
```

Comment:
Why return True (configuration is valid) when vdbench config is missing?

Reply:
Yes right , Raised  ValueError if vdbench_config is not defined.